libsolidity/codegen/MultiUseYulFunctionCollector.cpp

```diff
@@ -26,19 +26,39 @@
 #include <libsolutil/Whiskers.h>
 #include <libsolutil/StringUtils.h>
 
+#include <regex>
+
 using namespace std;
 using namespace solidity;
 using namespace solidity::frontend;
 using namespace solidity::util;
 
-string MultiUseYulFunctionCollector::requestedFunctions()
+string MultiUseYulFunctionCollector::requestedFunctions(string const& _sourceLocationComment)
 {
 	string result;
+	// std::map guarantees ascending order when iterating through its keys.
 	for (auto const& [name, code]: m_requestedFunctions)
 	{
 		solAssert(code != "<<STUB<<", "");
-		// std::map guarantees ascending order when iterating through its keys.
-		result += code;
+
+		// Insert source location comment if it doesn't exist already.
+		if (code.find("/// @src") == string::npos && !_sourceLocationComment.empty())
```

Comment:
I could put this in an extra function, though this is the only place where this is used... I feel it's alright here..

Reply:
I'm not sure we should put this logic here...

Reply:
My impression is that the MultiUseFunctionCollector should not concern itself with source locations. In this case, I think it would be better to put source locations in front of every single function, even if there are repetitions.